src/symmetric.rs

```diff
@@ -471,6 +654,147 @@ impl SymmetricAlgorithmKey {
     }
 }
 
+#[cfg(feature = "block-cipher")]
+pub use block_cipher;
+#[cfg(feature = "block-cipher")]
+pub use block_cipher_trait::BlockCipherKey;
+#[cfg(feature = "block-cipher")]
+mod block_cipher_trait {
+    use super::*;
+    use core::convert::TryFrom;
+
+    /// Helper struct that implements `block_cipher::BlockCipher` trait.
+    ///
+    /// Windows CNG API allows to create symmetric cipher keys with chaining
+    /// modes predefined. Because `BlockCipher` trait is only concerned with the
+    /// underlying cipher op (and which can be further used to compose chaining
+    /// mode on top of), we need to make sure that the inherent chaining mode is
+    /// set to ECB (which can be thought of as having no chaining mode).
+    pub struct BlockCipherKey<A: Algorithm, B: KeyBits> {
+        key: super::Key<A, B>,
+        _algo: PhantomData<A>,
+        _bits: PhantomData<B>,
+    }
+
+    impl<A: Algorithm, B: KeyBits> Clone for BlockCipherKey<A, B> {
+        fn clone(&self) -> Self {
+            BlockCipherKey {
+                key: self.key.clone(),
+                _algo: PhantomData,
+                _bits: PhantomData,
+            }
+        }
+    }
+
+    impl<A: Algorithm, B: KeyBits> BlockCipherKey<A, B> {
+        pub fn into_key(self) -> super::Key<A, B> {
+            self.key
+        }
+    }
+
+    impl<A: Algorithm, B: KeyBits> super::Key<A, B> {
+        /// Returns a helper type that implements [`block_cipher::BlockCipher`] trait.
+        ///
+        /// Returns `Ok` iff the underlying key is set to ECB chaining mode, see
```

Comment:
```suggestion
        /// Returns `Ok` if and only if the underlying key is set to ECB chaining mode, see
```

Reply:
Done in https://github.com/emgre/win-crypto-ng/pull/31/commits/ff58ce47ad650dc07021721b93fd42d9075f06c9.